new/collect/spec/index.spec.ts

```diff
@@ -0,0 +1,166 @@
+const mockWait = jest.fn(() => mockWaitResult);
+const mockLogResult = jest.fn();
+const mockLog = jest.fn(() => mockLogResult);
+const mockConfig = jest.fn();
+const mockMainPackage = jest.fn();
+const mockStartReport = jest.fn();
+const mockWhenNotDryrunResult = jest.fn();
+const mockWhenNotDryrun = jest.fn(() => mockWhenNotDryrunResult);
+// @TODO: Name this results properly
+const mockWhenNotStageResultResult = jest.fn();
+const mockWhenNotStageResult = jest.fn(() => mockWhenNotStageResultResult);
+const mockWhenNotStage = jest.fn(() => mockWhenNotStageResult);
+const mockCheckNpmPing = jest.fn();
+const mockCheckNpmLogin = jest.fn();
+const mockAddGitStatus = jest.fn();
+const mockCheckGitStatus = jest.fn();
+const mockReadStatusFile = jest.fn();
+const mockAddLastReleaseHash = jest.fn();
+const mockAddRawCommitMessages = jest.fn();
+const mockParseCommitMessages = jest.fn();
+const mockAddFilesToCommitMessages = jest.fn();
+
+jest.mock("../../helpers/compose-async", () => ({
+  composeAsync: mockComposeAsync,
+}));
+jest.mock("../../helpers/wait-module", () => ({ wait: mockWait }));
+jest.mock("../../helpers/log-module", () => ({ log: mockLog }));
+jest.mock("../config", () => ({ config: mockConfig }));
+jest.mock("../main-package", () => ({ mainPackage: mockMainPackage }));
+jest.mock("../start-report", () => ({ startReport: mockStartReport }));
+jest.mock("../../helpers/when", () => ({
+  whenNotDryrun: mockWhenNotDryrun,
+  whenNotStage: mockWhenNotStage,
+}));
+jest.mock("../check-npm", () => ({
+  checkNpmPing: mockCheckNpmPing,
+  checkNpmLogin: mockCheckNpmLogin,
+}));
+jest.mock("../add-git-status", () => ({ addGitStatus: mockAddGitStatus }));
+jest.mock("../check-git-status", () => ({
+  checkGitStatus: mockCheckGitStatus,
+}));
+jest.mock("../read-status-file", () => ({
+  readStatusFile: mockReadStatusFile,
+}));
+jest.mock("../add-last-release-hash", () => ({
+  addLastReleaseHash: mockAddLastReleaseHash,
+}));
+jest.mock("../add-raw-commit-messages", () => ({
+  addRawCommitMessages: mockAddRawCommitMessages,
+}));
+jest.mock("../parse-commit-messages", () => ({
+  parseCommitMessages: mockParseCommitMessages,
+}));
+jest.mock("../add-files-to-commit-messages", () => ({
+  addFilesToCommitMessages: mockAddFilesToCommitMessages,
+}));
+
+describe("collect's Index", () => {
+  let result: typeof composeAsyncResult;
+  beforeAll(() => {
+    result = require("../index").collect;
+  });
+
+  // Test Steps' order
+  test("composeAsync is called once", () => {
+    expect(mockComposeAsync).toBeCalledTimes(1);
+  });
+
+  test("1. logs a description", () => {
+    // @ts-ignore
+    expect(mockComposeAsync.mock.calls[0][0]).toBe(mockLogResult);
+    expect(mockLog).toHaveBeenCalledTimes(1);
+    expect(mockLog).toHaveBeenCalledWith("COLLECT PHASE: gathering all data");
+  });
+
+  test("2. reads config", () => {
+    // @ts-ignore
+    expect(mockComposeAsync.mock.calls[0][1]).toBe(mockConfig);
```

Comment:
What about using https://jestjs.io/docs/expect#tohavebeennthcalledwithnthcall-arg1-arg2-

`toHaveBeenNthCalledWith(1, mockConfig)`

So we don't rely on that ugly internal jest format.

Also for the others...

Reply:
I thought about `toHaveBeenNthCalledWith`, but then we'd have to check all arguments at once, while `.mock.calls` syntax allows us to check one by one the arguments

Reply:
👍  I'll close this conversation then.